bobweb/bob/test_nordpool_service.py

```diff
@@ -168,6 +164,70 @@ async def test_graph_with_narrower_width_is_as_expected(self):
         actual_array = (await get_data_for_date(today, graph_width=12)).data_graph
         self.assertEqual(actual_array, expected_array)
 
+    async def test_price_array_to_be_as_expected_when_missing_price_data(self):
+        """ If there is missing data in the data set it should be informed in the result and should not
+            cause an exception. """
```

Comment:
Not a good sign though that the API has missing data in the first place. I wonder if it's only for old past dates.

Reply:
By chance a date thet the tests were using (2023-02-17) had missing data on the new API response. It seems that the Nordpool API "fixed" this by returning same price as for the previous hour as with a quick check it was rare to have same exact price for 2 sequential hours. But now at least the bot knows and notifies user if the data is not exact.